sql/src/main/java/io/crate/executor/transport/ddl/TransportRenameTableAction.java

```diff
@@ -0,0 +1,154 @@
+
+package io.crate.executor.transport.ddl;
+
+import io.crate.metadata.PartitionName;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.apache.logging.log4j.util.Supplier;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.master.TransportMasterNodeAction;
+import org.elasticsearch.cluster.AckedClusterStateUpdateTask;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.ack.ClusterStateUpdateResponse;
+import org.elasticsearch.cluster.block.ClusterBlockException;
+import org.elasticsearch.cluster.block.ClusterBlockLevel;
+import org.elasticsearch.cluster.block.ClusterBlocks;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.MetaData;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.Priority;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.threadpool.ThreadPool;
+import org.elasticsearch.transport.TransportService;
+
+import java.util.Arrays;
+
+@Singleton
+public class TransportRenameTableAction extends TransportMasterNodeAction<RenameTableRequest, RenameTableResponse> {
+
+    private static final String ACTION_NAME = "crate/table/rename";
+    private static final IndicesOptions INDICES_OPTIONS = IndicesOptions.fromOptions(false, false, false, false);
+
+    @Inject
+    public TransportRenameTableAction(Settings settings,
+                                      TransportService transportService,
+                                      ClusterService clusterService,
+                                      ThreadPool threadPool,
+                                      ActionFilters actionFilters,
+                                      IndexNameExpressionResolver indexNameExpressionResolver) {
+        super(settings, ACTION_NAME, transportService, clusterService, threadPool, actionFilters,
+            indexNameExpressionResolver, RenameTableRequest::new);
+    }
+
+    @Override
+    protected String executor() {
+        // no need to use a thread pool, we go async right away
+        return ThreadPool.Names.SAME;
+    }
+
+    @Override
+    protected RenameTableResponse newResponse() {
+        return new RenameTableResponse();
+    }
+
+    @Override
+    protected void masterOperation(RenameTableRequest request,
+                                   ClusterState state,
+                                   ActionListener<RenameTableResponse> listener) throws Exception {
+        // validate the target table name
+        // done here, so we can tests that the source table will be re-opened also after rename failures
```

Comment:
can you please move "done here" to the previous line as it's confusing to read it as separate comments. Also there's a small type "so we can tests" -> "so we can test"